org.adempiere.base/src/org/compiere/print/DataEngine.java

```diff
@@ -1061,24 +1063,46 @@ else if (m_group.isFunctionColumn(pdc.getAD_PrintFormatItem_ID(), functions[f]))
 								{
 									/** START DEVCOFFEE: script column **/
 									int displayType = pdc.getDisplayType();
-									if (display.startsWith("@SCRIPT")) {
-										display = display.replace("@SCRIPT", "");
-										Object value = parseVariable(display, pdc, pd);
-										Interpreter bsh = new Interpreter ();
-										try {
-											value = bsh.eval(value.toString());
+									MPrintFormatItem pfItem = new MPrintFormatItem(Env.getCtx(), pdc.getAD_PrintFormatItem_ID(), null);
+									//if(rs.getMetaData().getColumnName(displayIndex).contains("SCRIPTCOLUMN")) {	// ScriptColumn
+									if(MPrintFormatItem.PRINTFORMATTYPE_Script.equalsIgnoreCase(pfItem.getPrintFormatType())) {	// ScriptColumn	
+										Object value = rs.getObject(displayIndex);
+										int columnType = rs.getMetaData().getColumnType(displayIndex);
+
+										// String, Text
+										if (columnType == Types.CHAR || columnType == Types.VARCHAR)
+										{
+											displayType = DisplayType.Text;
 										}
-										catch (EvalError e) {
-											log.severe(e.getMessage());
+										else if (columnType == Types.DATE || columnType == Types.TIME
+												|| columnType == Types.TIMESTAMP)
+										{
+											displayType = DisplayType.DateTime;
 										}
-										if (value instanceof Number)
+										// Number
+										else if (columnType == Types.INTEGER || columnType == Types.SMALLINT
+											|| columnType == Types.DECIMAL || columnType == Types.NUMERIC)
 											displayType = DisplayType.Number;
-										else if (value instanceof Boolean)
-											displayType = DisplayType.YesNo;
-										else if (value instanceof Date)
-											displayType = DisplayType.Date;
-										else
-											displayType = DisplayType.Text;
+									
+										if (display.startsWith("@SCRIPT")) {
+											display = display.replace("@SCRIPT", "");
+//											Object value = parseVariable(display, pdc, pd);
+											Interpreter bsh = new Interpreter ();
+											try {
+												value = bsh.eval(value.toString());
+											}
+											catch (EvalError e) {
+												log.severe(e.getMessage());
+											}
+											if (value instanceof Number)
+												displayType = DisplayType.Number;
+											else if (value instanceof Boolean)
+												displayType = DisplayType.YesNo;
+											else if (value instanceof Date)
+												displayType = DisplayType.Date;
+											else
+												displayType = DisplayType.Text;
```

Comment:
I think for the sake of consistency and simplicity, it is better to use the instanceof approach for both "@SCRIPT" and "@SQL"